app/src/main/java/com/weatherxm/data/datasource/DeviceOTADataSource.kt

```diff
@@ -1,8 +1,8 @@
 package com.weatherxm.data.datasource
 
 import arrow.core.Either
-import com.weatherxm.data.models.Failure
 import com.weatherxm.data.mapResponse
+import com.weatherxm.data.models.Failure
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/353#issuecomment-2392359926>#d0cda9</a><span> - 10/03/2024, 09:38 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The Failure model is imported but not consistently used across all methods in the DeviceOTADataSource interface.
</div>
<div id="fix">
<b>Fix:</b> Update all methods in the DeviceOTADataSource interface to consistently use Either&lt;Failure, T&gt; for error handling.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -9,14 +9,14 @@
  interface DeviceOTADataSource {
      suspend fun getFirmware(deviceId: String): Either<Failure, ByteArray>
 -    fun setDeviceLastOtaVersion(deviceId: String, otaVersion: String)
 +    fun setDeviceLastOtaVersion(deviceId: String, otaVersion: String): Either<Failure, Unit>
      fun getDeviceLastOtaVersion(deviceId: String): Either<Failure, String>
 -    fun setDeviceLastOtaTimestamp(deviceId: String)
 -    fun getDeviceLastOtaTimestamp(deviceId: String): Long
 +    fun setDeviceLastOtaTimestamp(deviceId: String): Either<Failure, Unit>
 +    fun getDeviceLastOtaTimestamp(deviceId: String): Either<Failure, Long>
  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged